filebeat/input/log/harvester.go

```diff
@@ -298,13 +348,37 @@ func (h *Harvester) Run() error {
 
 		// Update state of harvester as successfully sent
 		h.state = state
+
+		// Update metics of harvester as event was sent
+		h.metrics.readOffset.Set(state.Offset)
+		h.metrics.lastPublished.Set(time.Now())
+		h.metrics.lastPublishedEventTimestamp.Set(message.Ts)
+		h.checkSize <- struct{}{}
+	}
+}
+
+func (h *Harvester) monitorFileSize() {
+	for {
+		timer := time.NewTimer(30 * time.Second)
+		select {
+		case <-h.done:
+			return
+		case <-h.checkSize:
```

Comment:
what is the purpose of `h.checkSize`? You want to implement a fallthrough or just reset the timer?

Reply:
I wanted to implement `fallthrough`, and I failed to add the keyword. However, I don't think triggering size checking after each read event is required. It is not necessary to check file size after every event as Beats might not even emit metrics as frequently as a message is read. WDYT?